glib-macros/src/shared_boxed_derive.rs

```diff
@@ -218,6 +218,26 @@ pub fn impl_shared_boxed(input: &syn::DeriveInput) -> proc_macro2::TokenStream {
 
         #impl_from_value
 
+        impl #crate_ident::translate::FromGlibPtrBorrow<*const #refcounted_type_prefix::InnerType> for #name {
+            #[inline]
+            unsafe fn from_glib_borrow(ptr: *const #refcounted_type_prefix::InnerType) -> #crate_ident::translate::Borrowed<Self> {
+                debug_assert!(!ptr.is_null());
+
+                // from_raw is taking ownership of the raw pointer here, but wrapping its result
+                // in Borrowed::new ensures that it won't be deallocated when it will go out of
+                // scope, so the pointer will still be valid afterwards
```

Comment:
I'd do it consistently with the other one (using `ptr::read`), even if it's not strictly necessary here

Reply:
I think the `shared_boxed` case is a little different from the `boxed` one.
Our shared type is going to be something like:

```
MyShared(Arc<InnerType>)
```
or

```
MyShared(Rc<InnerType>)
```

so we actually need to get an `Arc` or an `Rc` from the raw pointer to the `InnerType` in order to get the `MyShared` object (in contrast to the `boxed` case in which we really needed just the `InnerType`). 

Another possibility would be wrapping the `InnerType` returned by `ptr::read` in a new `Arc` or `Rc`, but this sounds a bit strange to me, since the raw pointer actually comes from a different `Arc` or `Rc`, with its own already existing refcount

Reply:
Ah indeed. Hrm OK then! I don't like all this but I also don't see a better way :)